moveit_core/collision_detection_fcl/test/test_fcl_env.cpp

```diff
@@ -0,0 +1,307 @@
+
+  // Adding the box right in front of the robot hand
+  shapes::Shape* shape = new shapes::Box(0.1, 0.1, 0.1);
+  shapes::ShapeConstPtr shape_ptr(shape);
+
+  Eigen::Isometry3d pos{ Eigen::Isometry3d::Identity() };
+  pos.translation().x() = 0.43;
+  pos.translation().y() = 0;
+  pos.translation().z() = 0.55;
+  c_env_->getWorld()->addToObject("box", shape_ptr, pos);
+
+  c_env_->setLinkPadding("panda_hand", 0.08);
+  c_env_->checkRobotCollision(req, res, *robot_state_, *acm_);
+  ASSERT_TRUE(res.collision);
+  res.clear();
+
+  c_env_->setLinkPadding("panda_hand", 0.0);
+  c_env_->checkRobotCollision(req, res, *robot_state_, *acm_);
+  ASSERT_FALSE(res.collision);
+}
+/** \brief Continuous self collision checks are not supported yet by tesseract */
+TEST_F(CollisionDetectionEnvTest, DISABLED_ContinuousCollisionSelf)
+{
+  collision_detection::CollisionRequest req;
+  collision_detection::CollisionResult res;
+
+  robot_state::RobotState state1(robot_model_);
+  robot_state::RobotState state2(robot_model_);
+
+  setToHome(state1);
+  double joint2 = 0.15;
+  double joint4 = -3.0;
+  double joint5 = -0.8;
+  double joint7 = -0.785;
+  state1.setJointPositions("panda_joint2", &joint2);
+  state1.setJointPositions("panda_joint4", &joint4);
+  state1.setJointPositions("panda_joint5", &joint5);
+  state1.setJointPositions("panda_joint7", &joint7);
+  state1.update();
+
+  c_env_->checkSelfCollision(req, res, state1, *acm_);
+  ASSERT_FALSE(res.collision);
+  res.clear();
+
+  setToHome(state2);
+  double joint_5_other = 0.8;
+  state2.setJointPositions("panda_joint2", &joint2);
+  state2.setJointPositions("panda_joint4", &joint4);
+  state2.setJointPositions("panda_joint5", &joint_5_other);
+  state2.setJointPositions("panda_joint7", &joint7);
+  state2.update();
+
+  c_env_->checkSelfCollision(req, res, state2, *acm_);
+  ASSERT_FALSE(res.collision);
+  res.clear();
+
+  // c_env_->checkSelfCollision(req, res, state1, state2, *acm_);
+  ROS_INFO_STREAM("Continous to continous collisions are not supported yet, therefore fail here.");
+  ASSERT_TRUE(res.collision);
+  res.clear();
+}
+
+/** \brief Two similar robot poses are used as start and end pose of a continuous collision check. */
+TEST_F(CollisionDetectionEnvTest, ContinuousCollisionWorld)
+{
+  collision_detection::CollisionRequest req;
+  req.contacts = true;
+  req.max_contacts = 10;
+  collision_detection::CollisionResult res;
+
+  robot_state::RobotState state1(robot_model_);
+  robot_state::RobotState state2(robot_model_);
+
+  setToHome(state1);
+  state1.update();
+
+  setToHome(state2);
+  double joint_2{ 0.05 };
+  double joint_4{ -1.6 };
+  state2.setJointPositions("panda_joint2", &joint_2);
+  state2.setJointPositions("panda_joint4", &joint_4);
+  state2.update();
+
+  c_env_->checkRobotCollision(req, res, state1, state2, *acm_);
+  ASSERT_FALSE(res.collision);
+  res.clear();
+
+  // Adding the box which is not in collision with the individual states but with the casted one.
```

Comment:
nit: FCL at the moment isn't casting, it's doing discrete edge checks.

Reply:
right